### ISSUE TITLE
Validation accuracy remains 0.

### ISSUE THREAD

**Issue by KeironO** [anayebi/keras-extra#6]
Hey there,

I have been playing around with your keras-extra add on for awhile - but I have been unable to yield any decent results whilst using it as opposed to my standard libraries.

Below is my code:

``` python

def create_lcrn(number_of_timesteps, height, width):
    print "Creating the LRCN:"
    model = Sequential()

    # Convolutional Layers
    model.add(TimeDistributedConvolution2D(32, 3, 3, border_mode='same', input_shape=(number_of_timesteps, 1, height, width)))
    model.add(Activation('relu'))

    model.add(TimeDistributedMaxPooling2D(pool_size=(3, 3)))
    model.add(Dropout(0.25))

    model.add(TimeDistributedConvolution2D(64, 3, 3, border_mode='same'))
    model.add(Activation('relu'))

    model.add(TimeDistributedMaxPooling2D(pool_size=(3, 3)))
    model.add(Dropout(0.25))

    model.add(TimeDistributedFlatten())

    model.add(LSTM(128, return_sequences=True))
    model.add(Dropout(0.2))
    model.add(LSTM(64))

    model.add(Dense(2))
    model.add(Activation('softmax'))

    # o = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
    o = RMSprop(lr=0.0005, rho=0.9, epsilon=1e-06)
    model.compile(loss="binary_crossentropy", optimizer=o)
    print "\t Model Created!"
    return model

def train_model(model, train_data, train_label):
    train_label = np_utils.to_categorical(train_label, 2)
    model.fit(train_data, train_label, batch_size=2, nb_epoch=10, verbose=1, validation_split=0.2, show_accuracy=True)

if __name__ == "__main__":
    # Details for training
    number_of_folds = 2

    # Details for data
    number_of_timesteps, height, width = 53, 60, 80

    all_test_labels = []
    all_prediction_probabilities = []
    all_predictions = []


    data, labels = load_data(height, width, number_of_timesteps)
    number_of_samples = data.shape[0]

    print "\tVideo loaded in shape of:", data.shape
    print "\t\tNumber of samples:", number_of_samples, ", Number of timesteps:", number_of_timesteps


    stratified_k_fold = StratifiedKFold(labels, n_folds=number_of_folds, shuffle=True)
    for i, (train, test) in enumerate(stratified_k_fold):
        model = None
        model = create_lcrn(number_of_timesteps, height, width)
        train_model(model, data[train], labels[train])
```

And here is the output.

```
Using Theano backend.
Loading PoD video data:
    Video loaded in shape of: (52, 53, 1, 60, 80)
        Number of samples: 52 , Number of timesteps: 53
Creating the LRCN:
     Model Created!
Train on 20 samples, validate on 6 samples
Epoch 1/10
20/20 [==============================] - 55s - loss: 0.4085 - acc: 0.8000 - val_loss: 2.5858 - val_acc: 0.0000
Epoch 2/10
20/20 [==============================] - 55s - loss: 0.2131 - acc: 0.9500 - val_loss: 2.9949 - val_acc: 0.0000
Epoch 3/10
20/20 [==============================] - 55s - loss: 0.2075 - acc: 0.9500 - val_loss: 3.0842 - val_acc: 0.0000
```

Any help would be gratefully appreciated.

**Comment by KeironO** [comment]
Had a small problem with the loader, but even since it has been fixed I have been unable to effectively train a Neural Network!

**Comment by anayebi** [comment]
Some might say that the "dark art" of deep learning is that you have to try a lot of things before something starts to work. It could be an issue with whether you are randomizing your validation data, or your initialization, or your optimizer. I notice for the optimizer at least you are using rmsprop but not the default parameters for it. Default rmsprop works for fine me, as well as default adam, when I train a CNN-LSTM. But, of course, I'm not very familiar with your data, so it'll really come down to experimentation and tweaking, but those are important things to look for when you're starting out. (I do think most likely it has to do with a bug in how you're loading and processing your data than anything else).

**Comment by KeironO** [comment]
Quick update, you were correct. Some of my labels were incorrect, and as a result the NN was unable to distinguish between either class.
